### DISCUSSION TITLE
How to use `@vue/babel-plugin-jsx` in swc?

### DISCUSSION THREAD

**Discussion by Edge00** [swc-project/swc#4755] [Q&A] [answered]
### Describe the feature

I am using `swc-loader` to make compile quikcker and it works really nice.
However if a `.vue` file was written in `tsx`, swc-loader can't handle it.
And i notice that babel works well because this plugin: [@vue/babel-plugin-jsx](https://github.com/vuejs/babel-plugin-jsx).

Can swc provide such capacity? Or how can i make some contribution?

### Babel plugin or link to the feature description

https://github.com/vuejs/babel-plugin-jsx

**Comment by kwonoj** [comment]
Similar to the babel plugin you may try to write plugin for the swc.

https://github.com/swc-project/swc/discussions/3540.

**Comment by kwonoj** [comment]
This issue has been moved to the discussion.

**Comment by kwonoj** [comment, marked as answer]
Also: https://github.com/swc-project/swc/discussions/4768

**Comment by Austaras** [comment]
I've checked the implementation of vue jsx, it seems very easy to implement it using existing code -- however it would be awakward to expose api, currently many things related with jsx are named with `react-xxx`

**Reply by Edge00** [reply]
Thank you! it would be a huge improvement for vue dev experience. I am sorry that i can't giving useful idea for api name problem.
